### PR TITLE
MHP-2877: Adding TypeScript

### PR DESCRIPTION
**Pull request by bwtoptal**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
-    "parser": "babel-eslint",
+    "parser": "@typescript-eslint/parser",
     "parserOptions": {
-        "sourceType": "module"
+        "ecmaVersion": 6,
+        "sourceType": "module",
+        "ecmaFeatures": {
+            "experimentalObjectRestSpread": true,
+            "jsx": true,
+            "legacyDecorators": true
```
Comment (OzzieOrca) [inline .eslintrc.json]: We using these anywhere? I think you can kill this.

File: .eslintrc.json [hunk 2]
```diff
@@ -1,7 +1,13 @@
 {
-    "parser": "babel-eslint",
+    "parser": "@typescript-eslint/parser",
     "parserOptions": {
-        "sourceType": "module"
+        "ecmaVersion": 6,
+        "sourceType": "module",
+        "ecmaFeatures": {
+            "experimentalObjectRestSpread": true,
```
Comment (OzzieOrca) [inline .eslintrc.json]: This still needed? Idk that it's experimental anymore...

File: src/dashboard/components/Header/index.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ const Header = styled.h1`
     font-weight: 300;
     font-size: 30px;
     line-height: 38px;
-    color: ${({ theme }) => theme.colors.primary};
+    color: ${(props: any) => props.theme.colors.primary};
```
Comment (OzzieOrca) [inline src/dashboard/components/Header/index.ts]:
```suggestion
    color: ${({ theme }: { theme: typeof defaultTheme }) => theme.colors.primary};
```
might work, assuming that you import `defaultTheme.js` (might need to be converted to TS). https://emotion.sh/docs/typescript#define-a-theme might work to do it more globally.

**Comment by OzzieOrca** [comment]
Can you add an npm script like https://github.com/CruGlobal/missionhub-react-native/blob/develop/package.json#L20 to run the compiler as a linter and then add that to run on Travis. Looks like jest is unhappy.

File: src/dashboard/components/BarChart/index.tsx [hunk 4]
```diff
@@ -164,11 +172,11 @@ const BarChart = props => {
         updateFilter(newIndex, filter.type);
     };
 
-    const tooltip = options => {
+    const tooltip = (options: any) => {
         const { data } = options;
 
         return [
-            data.stages.map(row => (
+            data.stages.map((row: any) => (
```
[the remark this answers is not in the document]
Comment (bwtoptal) [reply src/dashboard/components/BarChart/index.tsx]: Actually, it's not...

File: src/dashboard/components/BarChart/index.tsx [hunk 5]
```diff
@@ -191,6 +199,7 @@ const BarChart = props => {
                     keys={keys}
                     indexBy={indexBy}
                     theme={theme.graph}
+                    // @ts-ignore
```
[the remark this answers is not in the document]
Comment (bwtoptal) [reply src/dashboard/components/BarChart/index.tsx]: Yeah... I'm also trying to use it only if I can't fix the issue.. I tried for a bit and it should actually be doable, but for some reason when I import an enum type `BarLayerType` from `@nivo/bar` it's undefined... not sure why :/

File: src/dashboard/components/BarChart/index.tsx [hunk 6]
```diff
@@ -217,6 +226,7 @@ const BarChart = props => {
                     axisTop={null}
                     groupMode={'grouped'}
                     axisRight={null}
+                    // @ts-ignore
```
[the remark this answers is not in the document]
Comment (bwtoptal) [reply src/dashboard/components/BarChart/index.tsx]: No sure how to convert my function to `TooltipProp`...

File: src/dashboard/insights.tsx [hunk 7]
```diff
@@ -9,7 +9,13 @@ import AppContext from './appContext';
 import defaultTheme from './defaultTheme';
 import Layout from './containers/Layout';
 
-const Insights = ({ orgId, authenticationService }) => {
+const Insights = ({
+    orgId,
+    authenticationService,
+}: {
+    orgId: number;
```
[the remark this answers is not in the document]
Comment (bwtoptal) [reply src/dashboard/insights.tsx]: You're right..

**Pull request merged**